docx/docx.go

```diff
@@ -23,66 +24,156 @@ type OpenControlDocx struct {
 }
 
-//BuildDocx exports a Doxc ssp based on a template
+//BuildDocx exports a Docx ssp based on a template
 func (config *Config) BuildDocx() error {
 	openControl := OpenControlDocx{models.LoadData(config.OpencontrolDir, "")}
 	docTemplate, err := docTemp.GetTemplate(config.TemplatePath)
 	if err != nil {
 		return err
 	}
-	funcMap := template.FuncMap{"getControl": openControl.FormatControl}
+	funcMap := template.FuncMap{
+		"getAllControlSections": openControl.FormatControl,
+		"getControlSection":     openControl.FormatControl,
+		"getParameter":          openControl.FormatParameter,
+		"getResponsibleRole":    openControl.FormatResponsibleRoles,
+	}
 	docTemplate.AddFunctions(funcMap)
 	docTemplate.Parse()
 	docTemplate.Execute(config.ExportPath, nil)
 	return err
 }
 
-// FormatControl returns a control formatted for docx
-func (openControl *OpenControlDocx) FormatControl(standardControl string) string {
+type componentInfoType int
+
+const (
+	noneInfo componentInfoType = iota
+	controlInfo
+	parameterInfo
+	responsibleRoleInfo
+)
+
+// createSectionsSet creates a set of section headers to do easy searching from the slice of string section keys
+func createSectionsSet(sections ...string) *set.Set {
+	sectionsSet := set.New()
+	for _, section := range sections {
+		sectionsSet.Add(section)
+	}
+	return sectionsSet
+}
+
+func getControlInfo(text string, justification models.Verification, component base.Component, specifiedSections *set.Set) (string, bool) {
+	// Add the component name.
+	text = fmt.Sprintf("%s%s\n", text, component.GetName())
+
+	// foundText is a placeholder to indicate that we actually found text for the section.
+	foundText := false
+
+	// Determine if we want to get all of the sections or just one. If we specify exact sections, that means we do
+	// not want all and if we do not specify sections, it means we want all sections.
+	allSections := specifiedSections.Size() == 0
+	// Print out the narrative(s)
+	for _, section := range justification.SatisfiesData.GetNarratives() {
+		if allSections {
+			// If we want to print out all the sections...
+
+			// If section header exists, let's print it. Key could be empty, in that case
+			// just print the text for the section.
+			if section.GetKey() != "" {
+				text = fmt.Sprintf("%s%s:\n", text, section.GetKey())
+			}
+			text = fmt.Sprintf("%s%s\n", text, section.GetText())
+
+			// Automatically assume foundText is true as long as the length of
+			// justification.SatisfiesData.Narrative is > 0, which is implied if we reach here.
+			// Also, in case the section in the YAML is explicitly "", we accept empty string here too.
+			foundText = true
+		} else {
+			// If we only want certain section(s)...
+
+			// If section header exists, let's print it's corresponding text and not the header itself.
+			if specifiedSections.Has(section.GetKey()) {
+				text = fmt.Sprintf("%s%s\n", text, section.GetText())
+				foundText = true
+			}
+		}
+	}
+	return text, foundText
+}
```

Comment:
This function is doing a lot...can we split it up? Example functions that could come out of it:

* `getCombinedControlInfo()`
* `getSectionInfo()`
* `getNarrative()`

Also, not sure that `foundText` is necessary, but it's not hurting anything I suppose.